swiftsimio/subset_writer.py

```diff
@@ -0,0 +1,224 @@
+"""
+Contains functions for reading a subset of a SWIFT dataset and writing
+it to a new file.
+"""
+
+from swiftsimio.reader import SWIFTUnits, SWIFTMetadata
+from swiftsimio.masks import SWIFTMask
+from swiftsimio.accelerated import read_ranges_from_file
+import swiftsimio.metadata as metadata
+
+import unyt
+import h5py
+import numpy as np
+
+
+def get_dataset_mask(mask: SWIFTMask, dataset_name: str, suffix=""):
```

Comment:
Please add type annotation for suffix. Also not comfortable using a string here as the default, as that's mutable.

So:

```
suffix: Optional[str] = None
```
You can get `Optional` from the `typing` module.

Then in the code, we can do `suffix = "" if suffix is None else suffix`